clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/HonoTopic.java

```diff
@@ -94,22 +116,27 @@ public int hashCode() {
      */
     public enum Type {
 
-        TELEMETRY(NAMESPACE + TelemetryConstants.TELEMETRY_ENDPOINT + SEPARATOR),
-        EVENT(NAMESPACE + EventConstants.EVENT_ENDPOINT + SEPARATOR),
-        COMMAND(NAMESPACE + CommandConstants.COMMAND_ENDPOINT + SEPARATOR),
-        COMMAND_RESPONSE(NAMESPACE + CommandConstants.COMMAND_RESPONSE_ENDPOINT + SEPARATOR);
+        TELEMETRY(TelemetryConstants.TELEMETRY_ENDPOINT),
+        EVENT(EventConstants.EVENT_ENDPOINT),
+        COMMAND(CommandConstants.COMMAND_ENDPOINT),
+        COMMAND_RESPONSE(CommandConstants.COMMAND_RESPONSE_ENDPOINT);
 
+        final String endpoint;
         final String prefix;
 
-        Type(final String prefix) {
-            this.prefix = prefix;
+        Type(final String endpoint) {
+            this.endpoint = endpoint;
+            this.prefix = NAMESPACE + SEPARATOR + endpoint + SEPARATOR;
```

Comment:
what is the prefix good for? is it being accessed by means of the field directly? If so then why do we have a *getEndpoint()* accessor? IMHO the fields should be accessed in a consistent way, i.e. either directly (no accessors) or via accessors (fields declared private) ...